src/App.tsx

```diff
@@ -1,24 +1,86 @@
 import React from 'react';
 import './App.scss';
 
-const App: React.FC = () => {
-  setInterval(() => {
-    const date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean;
+  clockName: number | string;
+  date: string;
 };
 
+class App extends React.Component<{}, State> {
+  state: State = {
+    isClockVisible: true,
+    clockName: 'New',
+    date: (new Date()).toLocaleTimeString(),
+  };
+
+  componentDidMount() {
+    setInterval(() => {
+      this.setState({ date: (new Date()).toLocaleTimeString() });
+    }, 1000);
+  }
+
+  hideClock = () => {
+    this.setState({ isClockVisible: false });
+  };
+
+  showClock = () => {
+    this.setState({ isClockVisible: true });
+  };
+
+  setName = () => {
+    this.setState((prevState) => {
+      const randomName = Math.ceil(Math.random() * 1000);
+
+      // eslint-disable-next-line no-console
+      console.log(`The Clock was renamed from ${prevState.clockName} to ${randomName}`);
+
+      return {
+        clockName: randomName,
+      };
+    });
+  };
+
+  render() {
+    return (
+      <div className="App">
+
+        {this.state.isClockVisible
+          ? (
```

Comment:
```suggestion
        {this.state.isClockVisible && (
```
Use this syntax of conditional rendering instead of ternaries